server/src/server.js

```diff
@@ -2,25 +2,53 @@ import dotenv from 'dotenv';
 import http from 'http';
 import cors from 'cors';
 import express from 'express';
-import morgan from 'morgan';
+import logger from 'morgan';
+import passport from 'passport';
+import session from 'express-session';
+import connectMongoDBSession from 'connect-mongodb-session';
+import cookieParser from 'cookie-parser';
 import connect from './schemas/index.js';
 import routes from './routes/index.js';
+import passportIndex from './middleware/index.js';
 
 dotenv.config();
 
-const PORT = process.env.PORT || 10010;
+const { MONGODB_HOST, MONGODB_USER, MONGODB_PW, DATABASE, SECRET_KEY } =
+  process.env;
 
-const allowedOrigins = [`http://localhost:${PORT}`];
+const PORT = process.env.PORT || 10010;
 
-const corsOption = {
-  origin: '*',
-};
+const MongoDBStore = connectMongoDBSession(session);
+const sessionStore = new MongoDBStore({
+  uri: `mongodb+srv://${MONGODB_USER}:${MONGODB_PW}@${MONGODB_HOST}/${DATABASE}?retryWrites=true&w=majority`,
+  collection: 'sessions',
+});
 
 connect();
 
 const app = express();
-app.use(cors(corsOption));
 app.use(express.json());
+app.use(logger('dev'));
+app.use(cookieParser());
+
+app.use(
+  session({
+    secret: `${SECRET_KEY}`,
+    resave: false,
+    saveUninitialized: false,
+    store: sessionStore,
+  })
+);
+app.use(passport.initialize());
+app.use(passport.authenticate('session'));
+passportIndex();
+
+const corsOptions = {
+  origin: 'http://localhost:3000',
```

Comment:
we need origin: '*'

Reply:
Setting the `origin` as '*' did not work. May I request to investigate it together?